### PR TITLE
Allow partial bonds claim

### PR DESCRIPTION
**Pull request by grandizzy**
- allow kicker to withdraw parts of claimable bonds amount
- add `amount` param to `withdrawBonds` function (The max amount to withdraw from auction bonds. Constrained by claimable amounts and liquidity)
- the amount to transfer is the min of passed amount parameter, claimable balance and quote token balance of the pool
- if amount to transfer is 0 function reverts with `InsufficientLiquidity`
- added new `BondWithdrawn` event containing owner address, recipient to receive amount address and amount
- added tests (didn't manage to replicate a scenario where liquidity / reserves of the pool less than what kicker is entitled to claim)

### CODE REVIEW

**Review by mattcushman — APPROVED** [review]
lgtm

File: src/base/Pool.sol [hunk 1]
```diff
@@ -424,14 +424,29 @@ abstract contract Pool is Clone, ReentrancyGuard, Multicall, IPool {
...
      *  @inheritdoc IPoolLiquidationActions
      *  @dev write state:
-     *       - reset kicker's claimable accumulator
+     *       - decrease kicker's claimable accumulator
+     *       - decrease auctions totalBondEscrowed accumulator
      */
-    function withdrawBonds(address recipient_) external {
+    function withdrawBonds(
+        address recipient_,
+        uint256 amount_
```
Comment (prat-gpt) [inline src/base/Pool.sol]: we can rename `amount_` to `maxAmount_` to be consistent with rest of the code.

**Review by prat-gpt — APPROVED** [review]
LGTM!  Just added one comment

File: src/base/Pool.sol (hunk 1 above)
Comment (grandizzy) [reply src/base/Pool.sol]: yep, will do

Comment (grandizzy) [reply src/base/Pool.sol]: changed with https://github.com/ajna-finance/contracts/pull/642/commits/7e5e546462ecdd96a214e4a3d805b173e154c7a7

**Pull request merged**